### ISSUE TITLE
Allow Skipping Initialization of Modules and Plugins for Dependencies

### ISSUE THREAD

**Issue by jason-huling** [gruntwork-io/terragrunt#950]
I have some terragrunt.hcl files with a few dependencies referenced within them (see https://github.com/gruntwork-io/terragrunt/issues/949). I noticed that for all of the dependencies, terragrunt will start "Initializing remote state" with the following:
```
Running command: terraform init -backend-config=bucket=<REDACTED> -backend-config=<REDACTED>
``` 
Then terraform will go through:
```
Initializing modules...
Initializing the backend...
Initializing provider plugins...
```
Is it possible for us to somehow pass through `-get=false` and `-get-plugins=false` to the `terraform init` command when initializing dependencies so that module and plugin initialization is skipped? Or do these need to be initialized for some reason?

**Comment by yorinasub17** [comment]
See https://github.com/gruntwork-io/terragrunt#auto-init, where it describes options for customizing the init call, as well as disabling the automatic init workflow.

**Comment by jason-huling** [comment]
Thanks for providing the link @yorinasub17. Unfortunately though, it doesn't look like there is a way to specify `extra_arguments`, to pass in the `-get=false` and `-get-plugins=false` flags, that are only applied when the module is being referenced as a dependency, unless I am missing something? So setting those flag would effect the dependencies but also the parent module consuming the dependencies, which is not desired in this case.

With that said, it does look like there may be a terraform limitation... I did try the following in my root terragrunt.hcl:
```
  extra_arguments "init_args" {
    commands = [
      "init"
    ]

    arguments = [
      "-get=false",
      "-get-plugins=false",
    ]
  }
```
And terraform returned the following errors:
```
There are some problems with the configuration, described below.

The Terraform configuration must be valid before initialization so that
Terraform can determine which modules and providers need to be installed.

Error: Module not installed

  on main.tf line 54:
  54: module "foo" {

This module is not yet installed. Run "terraform init" to install all modules
required by this configuration.
```
When I changed to...
```
    arguments = [
      "-get-plugins=false",
    ]
``` 
I then got the following error:
```
Initializing provider plugins...
Missing required providers.
```

So it looks like I misinterpreted these `-get*` flags as meaning it would skip initializing module and plugins, when it really just skips downloading them, but still tries to initialize.

It looks like this then falls out of scope of anything terragrunt can do.

Maybe there is something that can be done with `terraform state pull` or `terraform state show`... but I think this starts to overlap with https://github.com/gruntwork-io/terragrunt/issues/949, so I will close this one out. Thank you!
